### PR TITLE
Code refactor & linting

### PR DESCRIPTION
**Pull request by johnny243**
This PR has the following features:

- Integration with `rubocop`
- Integration with `houndci`
- Refactors some parts of the current codebase
- Lints the current codebase with Ruby's latest syntax

### CODE REVIEW

File: app/controllers/api/auth_controller.rb [hunk 1]
```diff
@@ -1,58 +1,64 @@
...
-    return if user == nil
-    mfa = user.items.where("content_type" => "SF|MFA", "deleted" => false).first
-    return mfa
+    return if user.nil?
+
+    user.items.where(content_type: 'SF|MFA', deleted: false).first
   end
 
-  def verify_mfa(email)
+  def verify_mfa(_email)
```
Comment (mobitar) [inline app/controllers/api/auth_controller.rb]: If `email` is unused we should just remove this argument and also update function invocations to not take a parameter.

File: app/controllers/api/auth_controller.rb [hunk 2]
```diff
@@ -1,58 +1,64 @@
 class Api::AuthController < Api::ApiController
-
   skip_before_action :authenticate_user, except: [:change_pw, :update]
 
   before_action do
     # current_user can still be nil by here.
     user = User.find_by_email(params[:email])
-    if user && user.locked_until && user.locked_until.future?
-      render :json => {error: {message: "Too many successive login requests. Please try your request again later."}}, :status => 423
+    if user&.locked_until && user.locked_until.future?
```
Comment (mobitar) [inline app/controllers/api/auth_controller.rb]: Can this be simplified to `if user&.locked_until&.future?`

File: app/controllers/api/items_controller.rb [hunk 3]
```diff
@@ -1,23 +1,22 @@
 class Api::ItemsController < Api::ApiController
-
   def sync_manager
-    if !@sync_manager
+    unless @sync_manager
       version = params[:api]
-      if version == "20190520"
-        @sync_manager = SyncEngine::V20190520::SyncManager.new(current_user)
+      @sync_manager = if version == '20190520'
```
Comment (mobitar) [inline app/controllers/api/items_controller.rb]: Should the if statement here be on the next line?

Comment (johnny243) [reply app/controllers/api/items_controller.rb]: Yes, it works too:

```
@sync_manager =
  if version == '20190520'
    SyncEngine::V20190520::SyncManager.new(current_user)
  else
    SyncEngine::V20161215::SyncManager.new(current_user)
  end
```

**Pull request merged**